### DISCUSSION TITLE
GitHub Enterprise

### DISCUSSION THREAD

**Discussion by glennawatson** [dotnet-foundation/Home#42] [General Discussions]
Thought I would start a discussion about the GitHub Enterprise problem for projects.

Is it needed? I imagine that it's needed for projects like MSFT?

Would it be useful to have projects decide?

To add to this discussion I know my project ReactiveUI would like to opt out but others might have different opinions?

**Comment by rprouse** [comment]
For projects that were moved into GitHub Enterprise and would like to move back, we are working on a plan to do that. Please bear with us while we work through the details.

**Comment by Perksey** [comment]
Just for the record: I fully support DNF having admin access to repos - this is important for project continuity should maintainers go AWOL. But that should be the minimum mandatory extent!

Thanks for all your work on improving some of these issues, sounds like a busy first week :D

**Reply by leastprivilege** [reply]
Isn't that a built-in feature of GitHub?

https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-user-account/managing-access-to-your-personal-repositories/maintaining-ownership-continuity-of-your-user-accounts-repositories

**Reply by glennawatson** [reply]
Nice find, wasn't aware of that feature.

**Comment by grokys** [comment]
I would like to get some clarification about what the Enterprise instance means practically. We use the https://github.com/AvaloniaUI/ org to host some private repos which we're thinking of turning into commercial products. Does the enterprise org mean the the dnf org owns these repositories now? How does that affect our plans?

A clear explanation of the reasoning for creating the enterprise org, and the ramifications of it would be greatly appreciated.

**Reply by richlander** [reply]
We will post on the .NET Foundation use of GitHub Enterprise shortly (today or Monday). We're going to work with maintainers to ensure that they are happy and can move forward with their plans w/o issue.

**Reply by leastprivilege** [reply]
any update? thanks!

**Reply by rprouse** [reply]
> any update? thanks!

Sorry we're late on this, our day jobs got in the way of completing it and getting it out on time. We're working on it.

**Reply by leastprivilege** [reply]
no problem, thanks!

**Reply by glennawatson** [reply]
Btw you can convert your org back to public org. 

Rename your org to dnf-[project name] then transfer the projects across. You'll loose shared GitHub secrets, some settings and teams.

**Reply by leastprivilege** [reply]
I don't think this is the right way to do it.

**Reply by glennawatson** [reply]
Yeah just some maintainers have taken this approach

**Reply by dansiegel** [reply]
At this point I think it's really not necessary @glennawatson... really best to respect that the board has said they are working on this and give them the time to follow through. Certainly with an understanding that that timeframe should be reasonable... not something we should be still talking about a month from now. 😉

**Reply by rprouse** [reply]
Give us a bit, we are working on plans to make it easier to transfer back, you shouldn't have to jump through hoops or lose settings.

**Comment by rprouse** [comment]
I have posted the board's response to the GHE issue here, https://github.com/dotnet-foundation/Home/discussions/59. It is not intended to usurp this thread, but we wanted the official position to be a top-level announcement.

**Reply by jglozano** [reply]
Like always, we would love to hear thoughts on the subject.
